### ISSUE TITLE
[$500] Clean up underscore and lodash imports in Policy.js

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#30854]
# Problem
We have several imports for `underscore` and `lodash` in the Policy.js file.

# Why is this important
Keeps code clean

# Solution
Remove extra imports
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a3cbec68a58fc514</li>
        <li>Upwork Job ID: 1720487717757710336</li>
        <li>Last Price Increase: 2023-11-03</li>
<li>Automatic offers: </li>
<ul>
<li>alitoshmatov | Reviewer | 27564915</li>
<li>DylanDylann | Contributor | 27564918</li>
</ul></li>
    </ul>
</details>

**Comment by DylanDylann** [comment]
@luacmartins Could you help to point out which exactly file (include path) ? I find Policy.ts on lastest main and don't see any import underscore or lodash

<img width="1440" alt="Screenshot 2023-11-03 at 22 43 01" src="https://github.com/Expensify/App/assets/141406735/6e6fffb8-2e49-4ae0-8900-990dac53dbe0">

**Comment by luacmartins** [comment]
Sorry, I meant Policy.js - https://github.com/Expensify/App/blob/c975ed9378c147daa104ad7e4ee3fc28ac7ad109/src/libs/actions/Policy.js#L3-L5

**Comment by DylanDylann** [comment]
@luacmartins You mean that we clean both all lodash and underscore import ? I wonder which one will we use to replace them?

**Comment by luacmartins** [comment]
We're using only lodash going forward. I also see a bunch of separate imports like the ones below:

```
import {escapeRegExp} from 'lodash';
import lodashGet from 'lodash/get';
import lodashUnion from 'lodash/union';
```

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Clean up underscore and lodash imports in Policy.js

### What is the root cause of that problem?
We have a plan to use lodash only so we need to clear underscore import  

### What changes do you think we should make in order to solve the problem?
1. We should remove this line
```
import _ from 'underscore';
```
and use lodash instead of

2. If we want to import lodash in a shorter way
we can remove separate imports
```
import lodashGet from 'lodash/get';
import lodashUnion from 'lodash/union';
import lodashUnion from 'lodash/union';
```
and use this inport
```
import {escapeRegExp, filter, get, union} from 'lodash';
```
or 
```
import _ from 'lodash';
```
### What alternative solutions did you explore? (Optional)

**Comment by just4kick** [comment]
# Proposal
## issue we are trying to solve #30854
 * Change underscore import to lodash import and merge separate lodash import into singal lodash import. 
## Changes required to solve the issue 
we have to remove ``` import _ from 'underscore';``` and replace with
```
import _ from 'lodash';
```
these separate import can be
```  
import {escapeRegExp} from 'lodash';
import lodashGet from 'lodash/get';
import lodashUnion from 'lodash/union'; 
```   
replace with

```
import { 
    get as lodashGet, 
    union as lodashUnion,
    escapeRegExp
} from 'lodash';
```

**Comment by just4kick** [comment]
Contributor details
Your Expensify account email: sumitshar16@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01a210def7b823c893

**Comment by just4kick** [comment]
# Proposal
## issue we are trying to solve #30854
 * Change underscore import to lodash import and merge separate lodash import into singal lodash import. 
## Changes required to solve the issue 
we have to remove ``` import _ from 'underscore';``` and replace with
```
import _ from 'lodash';
```
these separate import can be
```  
import {escapeRegExp} from 'lodash';
import lodashGet from 'lodash/get';
import lodashUnion from 'lodash/union'; 
```   
replace with

```
import { 
    get as lodashGet, 
    union as lodashUnion,
    escapeRegExp
} from 'lodash';
```

**Comment by tudorilisoi** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

We have several imports for underscore and lodash in the Policy.js file.
underscore is used in another ~340 files, found by searching `import _ from 'underscore'` in the project folder

### What is the root cause of that problem?

Lodash and underscore are both used which makes (obsolete package) underscore redundant.
There is no point in using per package imports such as [1]
```
import lodashGet from 'lodash/get';
```
Importing single packages like in the above example is supposed to minimize the webpack bundle size.


However,  the whole lodash is used across the project in about 14 files: `import _ from 'lodash'` [2], so being pedantic with single package imports  only in the `Policy.js` file does not amount to a smaller webpack bundle.

See also https://github.com/facebook/metro/issues/227#issuecomment-583358386 [3] about tree-shaking, the technique used to minimize the webpack bundle. 

In short, [1] has no sense in the context of [2] and [3].

### What changes do you think we should make in order to solve the problem?

 - replace **all** underscore imports  
 `import _ from 'underscore'` 
with 
`import _ from 'lodash'` 
**across the project**
 - replace per-package imports such as 
 `import lodashGet from 'lodash/get'` 
with a single 
`import _ from 'lodash'` 
, and change the code accordingly, e.g. replacing `lodashGet(...)` with `_.get(...)`.
This will make the code more concise, standardized and aligned with lodash documentation.


### What alternative solutions did you explore? (Optional)

none

**Comment by tudorilisoi** [comment]
Contributor details
Your Expensify account email: tudorilisoi@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01c54f0091dfb47bde

**Comment by luacmartins** [comment]
Waiting on @alitoshmatov to review proposals

**Comment by alitoshmatov** [comment]
Reviewing

**Comment by alitoshmatov** [comment]
The issue is straightforward and requires simple solution. Since all of the proposals are the same we can go with @DylanDylann 's [proposal](https://github.com/Expensify/App/issues/30854#issuecomment-1792788376). 

C+ reviewed 🎀 👀 🎀

**Comment by alitoshmatov** [comment]
One thing to consider is to which type of import to use, I am going to start a discussion in the slack

**Comment by luacmartins** [comment]
Actually, we will work on this once we [migrate the file to TS](https://expensify.slack.com/archives/C01GTK53T8Q/p1699388522008829). So I'm gonna close this issue. Thanks everyone for the proposals!
